sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfiguration.java

```diff
@@ -132,32 +130,34 @@ static MetricExporter configureOtlpMetrics(
       throw new ConfigurationException("Unsupported OTLP metrics protocol: " + protocol);
     }
 
-    configureIntervalMetricReader(config, meterProvider, exporter);
+    configureIntervalMetricReader(config, sdkMeterProviderBuilder, exporter);
 
     return exporter;
   }
 
   private static void configureIntervalMetricReader(
-      ConfigProperties config, SdkMeterProvider meterProvider, MetricExporter exporter) {
-    IntervalMetricReaderBuilder readerBuilder =
-        IntervalMetricReader.builder()
-            .setMetricProducers(Collections.singleton(meterProvider))
-            .setMetricExporter(exporter);
+      ConfigProperties config,
+      SdkMeterProviderBuilder sdkMeterProviderBuilder,
+      MetricExporter exporter) {
+
     Duration exportInterval = config.getDuration("otel.imr.export.interval");
-    if (exportInterval != null) {
-      readerBuilder.setExportIntervalMillis(exportInterval.toMillis());
+    if (exportInterval == null) {
+      exportInterval = Duration.ofMinutes(1);
     }
-    IntervalMetricReader reader = readerBuilder.build().startAndRegisterGlobal();
-    Runtime.getRuntime().addShutdownHook(new Thread(reader::shutdown));
+    // Register the reader (which starts it), and use a global shutdown hook rather than
+    // interval reader specific.
+    sdkMeterProviderBuilder.register(new PeriodicMetricReader.Factory(exporter, exportInterval));
   }
 
   private static void configurePrometheusMetrics(
-      ConfigProperties config, SdkMeterProvider meterProvider) {
+      ConfigProperties config, SdkMeterProviderBuilder sdkMeterProviderBuilder) {
     ClasspathUtil.checkClassExists(
         "io.opentelemetry.exporter.prometheus.PrometheusCollector",
         "Prometheus Metrics Server",
         "opentelemetry-exporter-prometheus");
-    PrometheusCollector.builder().setMetricProducer(meterProvider).buildAndRegister();
+    sdkMeterProviderBuilder.register(PrometheusCollector.create());
+    // TODO: Can we move this portion into the PrometheusCollector so shutdown on SdkMeterProvider
+    // will collapse prometheus too?
```

Comment:
Not the config reading part, but the server creation/startup makes sense to move there, yes.